src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -120,6 +120,13 @@ public partial interface IAzureResource : IResource
 
 namespace Aspire.Hosting.Azure
 {
+    public partial class AppliedRoleAssignmentsAnnotation : ApplicationModel.IResourceAnnotation
+    {
+        public AppliedRoleAssignmentsAnnotation(System.Collections.Generic.HashSet<RoleDefinition> roles) { }
```

Comment:
ISet? @eerhardt

Reply:
I wanted someone else's eyes on how I'm exposing these collections across these new annotations. Here's what we have today:

* `class AppliedRoleAssignmentsAnnotation`
    * `HashSet<RoleDefinition> Roles`
* `class RoleAssignmentAnnotation`
    * `IReadOnlySet<RoleDefinition> Roles`
* `class DefaultRoleAssignmentsAnnotation`
    * `IReadOnlySet<RoleDefinition> Roles`

The reason I made `AppliedRoleAssignmentsAnnotation` mutable is because we loop over the RoleAssignmentAnnotations and add them into the singular `AppliedRoleAssignmentsAnnotation`. That allows for simplier code in the resources - they only need to get 1 AppliedRoleAssignmentsAnnotation and not loop over them all. The other 2 feel more like read only is appropriate.

Reply:
This ones need more explanation. I think I understand the second and third bullet, what is AppliedRoleAssignmentsAnnotation?

Reply:
AppliedRoleAssignmentsAnnotation has been removed. #8280